lib/backup_restore/logger.rb

```diff
@@ -27,6 +27,30 @@ def log(message, ex = nil)
       end
     end
 
+    def self.save_log_to_upload(user:, filename: 'log.txt', logs:)
+      Dir.mktmpdir do |dir|
+        logfile = File.new(File.join(dir, filename), 'w')
+        logfile.write(Discourse::Utils.pretty_logs(logs))
+        logfile.close
+
+        zipfile = Compression::Zip.new.compress(dir, filename)
+        upload = File.open(zipfile) do |file|
+          UploadCreator.new(
```

Comment:
Just curious, did we consider what would happen if the file is greater than `SiteSetting.max_attachment_size_kb`?

Reply:
Validation will fail, upload will not persist and that case is handled outside this function call.